### DISCUSSION TITLE
Three same (uninformative) error messages due to uncompilable source code

### DISCUSSION THREAD

**Discussion by mareq** [neoclide/coc.nvim#3022] [Q&A]
The `rust-analyzer` reports and error about failing `cargo check` (there is intentionally introduced error by commenting out one of the `use` clauses, so there is an undefined symbol referred to in the code). I have written up [detailed description of the error](https://github.com/rust-analyzer/rust-analyzer/issues/3155#issuecomment-817403059) in the `rust-analyzer` repository.

The problem is, that this results in three identical floating windows popping up (or alternatively, the same error message consisting of many lines is echoed three times over, requiring hitting Enter to continue).

Is this works-as-designed, or is there some less intrusive possibility of reporting this kind of error? I guess the error needs to be reported somehow and it is quite possible that the actual problem is on the `rust-analyzer` side? Still, would there be a way to suppress the error, or at least not reporting it three times over?

**Comment by fannheyward** [comment]
Do you use `coc-rust-analyzer` or set languangeserver rust-analyzer in coc-settings.json? Can you post an image of the three error messages?

**Reply by mareq** [reply]
I use `coc-rust-analyzer` installed using `coc-marketplace`.

The popups look like this (but please see [my post in `rust-analyzer`](https://github.com/rust-analyzer/rust-analyzer/issues/3155#issuecomment-817403059) for exact details) and it was reproduced by directly opening the displayed file (`nvim backend/src/...rs`):
![image](https://user-images.githubusercontent.com/236290/114328288-089c5580-9b34-11eb-976d-184774891176.png)

(My screen is much bigger than this - I made the terminal window quite small, so that I do not produce huge image - so the effect is not by far that dramatic, but it still would be nice if to somehow not have those errors popping up.)


My configuration for `rust-analyzer` looks like this ATM, but this behaviour does not seem to change at all by changing this config:
```
  "rust-analyzer.enable": true,
  "rust-analyzer.updates.prompt": true,
  "rust-analyzer.updates.channel": "nightly",
  "rust-analyzer.lruCapacity": 16384,

  "rust-analyzer.cargo.runBuildScripts": true,
  "rust-analyzer.checkOnSave.enable": true,
  "rust-analyzer.checkOnSave.allTargets": true,
  "rust-analyzer.checkOnSave.overrideCommand": null,
  "rust-analyzer.checkOnSave.command": "check",
  "rust-analyzer.checkOnSave.extraArgs": ["--verbose"],

  "rust-analyzer.rustfmt.overrideCommand": null,
  "rust-analyzer.rustfmt.extraArgs": [],

  "rust-analyzer.inlayHints.enable": true,
  "rust-analyzer.inlayHints.typeHints": true,
  "rust-analyzer.inlayHints.typeHintsSeparator": "     ",
  "rust-analyzer.inlayHints.chainingHints": true,
  "rust-analyzer.inlayHints.chainingHintsSeparator": "     ",
  "rust-analyzer.inlayHints.typeHintsWithVariable": true,
  "rust-analyzer.inlayHints.refreshOnInsertMode": true,

  "rust-analyzer.lens.enable": true,
  "rust-analyzer.lens.run": true,
  "rust-analyzer.lens.implementation": true,
  "rust-analyzer.lens.methodReferences": true,
```

**Reply by mareq** [reply]
The vim configuration for CoC plugin (apart of the JSON file) looks like this: https://gist.github.com/mareq/25fa15abbde7d1114f00c25b677e7560

I guess the only relevant part is the one that makes it display the error in pop-ups instead of echoing them (it will be one of the lines below the line 62 if I remember correctly).

**Reply by mareq** [reply]
Also (only somewhat related to the issue at hand), I do not seem to be able to pass that `--verbose` argument, the error messages refer to (attempt can be seen in my JSON config above; I have also tried the `overrideCommand`, but that did not help neither). Is there a way?

**Reply by fannheyward** [reply]
Have no idea about this. Can you reproduce this error with VSCode + rust-analyzer extension?

**Reply by mareq** [reply]
Unfortunately, I do not have VSCode - never had... :shrug:

**Reply by mareq** [reply]
To close the loop: the issue is being [solved](https://github.com/rust-analyzer/rust-analyzer/issues/3155#issuecomment-817814135) on the other side:
- https://github.com/rust-analyzer/rust-analyzer/pull/8476
- https://github.com/rust-analyzer/rust-analyzer/pull/8478
